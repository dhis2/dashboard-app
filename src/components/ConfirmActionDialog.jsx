import {
    Button,
    Modal,
    ModalContent,
    ModalActions,
    ButtonStrip,
    ModalTitle,
} from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import classes from './styles/ConfirmActionDialog.module.css'

const ConfirmActionDialog = ({
    onConfirm,
    onCancel,
    open,
    title,
    message,
    cancelLabel,
    confirmLabel,
}) => {
    return (
        open && (
            <Modal
                dataTest="confirm-action-dialog"
                onClose={onCancel}
                small
                position="middle"
            >
                <ModalTitle>{title}</ModalTitle>
                <ModalContent>
                    <span className={classes.content}>{message}</span>
                </ModalContent>
                <ModalActions>
                    <ButtonStrip>
                        <Button key="cancel" secondary onClick={onCancel}>
                            {cancelLabel}
                        </Button>

                        <Button key="confirm" destructive onClick={onConfirm}>
                            {confirmLabel}
                        </Button>
                    </ButtonStrip>
                </ModalActions>
            </Modal>
        )
    )
}

ConfirmActionDialog.propTypes = {
    cancelLabel: PropTypes.string,
    confirmLabel: PropTypes.string,
    message: PropTypes.string,
    open: PropTypes.bool,
    title: PropTypes.string,
    onCancel: PropTypes.func,
    onConfirm: PropTypes.func,
}

export default ConfirmActionDialog
