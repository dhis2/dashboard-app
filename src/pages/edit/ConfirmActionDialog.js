import React from 'react'
import PropTypes from 'prop-types'
import i18n from '@dhis2/d2-i18n'
import {
    Button,
    Modal,
    ModalContent,
    ModalActions,
    ButtonStrip,
    ModalTitle,
} from '@dhis2/ui'

import classes from './styles/ConfirmActionDialog.module.css'

export const ACTION_DELETE = 'delete'
export const ACTION_DISCARD = 'discard'

const ConfirmActionDialog = ({
    action,
    dashboardName,
    onConfirm,
    onCancel,
    open,
}) => {
    const texts = {
        [ACTION_DELETE]: {
            title: i18n.t('Delete dashboard'),
            message: i18n.t(
                'Deleting dashboard "{{ dashboardName }}" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?',
                { dashboardName }
            ),
            cancel: i18n.t('Cancel'),
            confirm: i18n.t('Delete'),
        },
        [ACTION_DISCARD]: {
            title: i18n.t('Discard changes'),
            message: i18n.t(
                'This dashboard has unsaved changes. Are you sure you want to leave and discard these unsaved changes?'
            ),
            cancel: i18n.t('No, stay here'),
            confirm: i18n.t('Yes, discard changes'),
        },
    }

    const actions = [
        <Button key="cancel" secondary onClick={onCancel}>
            {texts[action].cancel}
        </Button>,
        <Button key="confirm" destructive onClick={onConfirm}>
            {texts[action].confirm}
        </Button>,
    ]

    return (
        open && (
            <Modal
                dataTest="confirm-action-dialog"
                onClose={onCancel}
                small
                position="middle"
            >
                <ModalTitle>{texts[action].title}</ModalTitle>
                <ModalContent>
                    <span className={classes.content}>
                        {texts[action].message}
                    </span>
                </ModalContent>
                <ModalActions>
                    <ButtonStrip>{actions}</ButtonStrip>
                </ModalActions>
            </Modal>
        )
    )
}

ConfirmActionDialog.propTypes = {
    action: PropTypes.string,
    dashboardName: PropTypes.string,
    open: PropTypes.bool,
    onCancel: PropTypes.func,
    onConfirm: PropTypes.func,
}

export default ConfirmActionDialog
