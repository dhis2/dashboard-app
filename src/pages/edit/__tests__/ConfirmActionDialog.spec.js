/* eslint-disable react/prop-types */
import React from 'react'
import { render } from '@testing-library/react'
import ConfirmActionDialog, {
    ACTION_DELETE,
    ACTION_DISCARD,
} from '../ConfirmActionDialog'

jest.mock('@dhis2/ui', () => {
    const originalModule = jest.requireActual('@dhis2/ui')

    return {
        __esModule: true,
        ...originalModule,
        Modal: function MockComponent(props) {
            return <div>{props.children}</div>
        },
    }
})

test('ConfirmActionDialog renders confirm delete dialog', () => {
    const { container } = render(
        <ConfirmActionDialog
            action={ACTION_DELETE}
            dashboardName="Twilight Sparkle"
            onConfirm={jest.fn()}
            onCancel={jest.fn()}
            open={true}
        />
    )
    expect(container).toMatchSnapshot()
})

test('ConfirmActionDialog renders discard changes dialog', () => {
    const { container } = render(
        <ConfirmActionDialog
            action={ACTION_DISCARD}
            onConfirm={jest.fn()}
            onCancel={jest.fn()}
            open={true}
        />
    )
    expect(container).toMatchSnapshot()
})

test('ConfirmActionDialog does not render dialog if not open', () => {
    const { container } = render(
        <ConfirmActionDialog
            action={ACTION_DISCARD}
            onConfirm={jest.fn()}
            onCancel={jest.fn()}
            open={false}
        />
    )
    expect(container).toMatchSnapshot()
})
