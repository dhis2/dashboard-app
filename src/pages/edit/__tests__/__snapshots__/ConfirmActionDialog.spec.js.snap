// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmActionDialog does not render dialog if not open 1`] = `<div />`;

exports[`ConfirmActionDialog renders confirm delete dialog 1`] = `
<div>
  <div>
    <h1
      class="jsx-3701517325 title"
      data-test="dhis2-uicore-modaltitle"
    >
      Delete dashboard
    </h1>
    <div
      class="jsx-659511263 "
      data-test="dhis2-uicore-modalcontent"
    >
      <span
        class="content"
      >
        Deleting dashboard "Twilight Sparkle" will remove it for all users. This action cannot be undone. Are you sure you want to permanently delete this dashboard?
      </span>
    </div>
    <div
      class="jsx-752323619"
      data-test="dhis2-uicore-modalactions"
    >
      <div
        class="jsx-71743532 start"
        data-test="dhis2-uicore-buttonstrip"
      >
        <div
          class="jsx-71743532 box"
        >
          <button
            class="jsx-2371629422 secondary"
            data-test="dhis2-uicore-button"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div
          class="jsx-71743532 box"
        >
          <button
            class="jsx-2371629422 destructive"
            data-test="dhis2-uicore-button"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConfirmActionDialog renders discard changes dialog 1`] = `
<div>
  <div>
    <h1
      class="jsx-3701517325 title"
      data-test="dhis2-uicore-modaltitle"
    >
      Discard changes
    </h1>
    <div
      class="jsx-659511263 "
      data-test="dhis2-uicore-modalcontent"
    >
      <span
        class="content"
      >
        This dashboard has unsaved changes. Are you sure you want to leave and discard these unsaved changes?
      </span>
    </div>
    <div
      class="jsx-752323619"
      data-test="dhis2-uicore-modalactions"
    >
      <div
        class="jsx-71743532 start"
        data-test="dhis2-uicore-buttonstrip"
      >
        <div
          class="jsx-71743532 box"
        >
          <button
            class="jsx-2371629422 secondary"
            data-test="dhis2-uicore-button"
            type="button"
          >
            No, stay here
          </button>
        </div>
        <div
          class="jsx-71743532 box"
        >
          <button
            class="jsx-2371629422 destructive"
            data-test="dhis2-uicore-button"
            type="button"
          >
            Yes, discard changes
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
